tests/benchmarks/android_camera_preview.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+
+import os
+from time import sleep
+
+# The workload class MUST be loaded before the LisaBenchmark
+from android import Workload
+from android import LisaBenchmark
+
+from devlib.exception import TargetError
+
+class CameraPreviewTest(LisaBenchmark):
+
+    # Android Workload to run
+    bm_name = 'CameraPreview'
+
+    # Default products to be collected
+    bm_collect = 'ftrace energy'
+
+    def benchmarkInit(self):
+        self.setupWorkload()
+        self.setupGovernor()
+
+    def __init__(self, governor, duration_s):
+        self.governor = governor
+        self.duration_s = duration_s
+        super(CameraPreviewTest, self).__init__()
+
+    def setupWorkload(self):
+        # Create a results folder for each "governor/test"
+        self.out_dir = os.path.join(self.te.res_dir, governor)
+        try:
+                os.stat(self.out_dir)
+        except:
+                os.makedirs(self.out_dir)
+        # Setup workload parameters
+        self.bm_params = {
+            'duration_s' : self.duration_s,
+        }
+
+    def setupGovernor(self):
+        try:
+            self.target.cpufreq.set_all_governors(self.governor);
+        except TargetError:
+            self._log.warning('Governor [%s] not available on target',
+                             self.governor)
+            raise
+
+        # Setup schedutil parameters
+        if self.governor == 'schedutil':
+            rate_limit_us = 2000
```

Comment:
Why not leaving this value as-is ? I'm not saying this is bad, and I can also see that we do the same for other benchmarks but what is the rationale ?